File: admin/src/api/interface/systemOperateLog.ts
// system_operate_log 操作日志
import { ReqPage } from "./index";
export namespace SystemOperateLog {
  export interface ReqSystemOperateLogList extends ReqPage {
    deleted?: number; // 删除
    username?: string; // 用户账号
    module?: string; // 模块名称
    beginStartTime?: string; // 开始操作时间
    finishStartTime?: string; // 结束操作时间
    result?: number; // 结果(0 成功/1 失败)
    deptId?: number; // 部门
  }
  export interface ResSystemOperateLogItem {
    id: number; // 主键
    username: string; // 用户账号
    module: string; // 模块名称
    requestMethod: string; // 请求方法名
    requestUrl: string; // 请求地址
    userIp: string; // 用户 ip
    userAgent: string | undefined; // UA
    goMethod: string; // 方法名
    goMethodArgs: any | undefined; // 方法的参数
    startTime: string; // 操作开始时间
    duration: number; // 执行时长
    channel: string; // 渠道
    result: number; // 结果(0 成功/1 失败)
    userId: number; // 用户 ID
    deleted: number; // 删除
    tenantId: number; // 租户
    creator: string | undefined; // 创建人
    createTime: string | undefined; // 创建时间
    updater: string | undefined; // 更新人
    updateTime: string | undefined; // 更新时间
  }
  export interface ReqSystemOperateLogMultiple {
    ids?: number[]; // 访问日志编号
  }
}

File: cloud/dao/system_operate_log.go
package dao

import "github.com/abulo/ratel/v3/stores/null"

// SystemOperateLog 操作日志 system_operate_log
type SystemOperateLog struct {
	Id            *int64        `db:"id" json:"id"`                        //bigint 主键,PRI
	Username      *string       `db:"username" json:"username"`            //varchar 用户账号
	UserId        *int64        `db:"user_id" json:"userId"`               //bigint 用户id
	Module        *string       `db:"module" json:"module"`                //varchar 模块名称
	RequestMethod *string       `db:"request_method" json:"requestMethod"` //varchar 请求方法名
	RequestUrl    *string       `db:"request_url" json:"requestUrl"`       //varchar 请求地址
	UserIp        *string       `db:"user_ip" json:"userIp"`               //varchar 用户 ip
	UserAgent     null.String   `db:"user_agent" json:"userAgent"`         //varchar UA
	GoMethod      *string       `db:"go_method" json:"goMethod"`           //varchar 方法名
	GoMethodArgs  null.JSON     `db:"go_method_args" json:"goMethodArgs"`  //json 方法的参数
	StartTime     null.DateTime `db:"start_time" json:"startTime"`         //datetime 操作开始时间
	Duration      *int32        `db:"duration" json:"duration"`            //int 执行时长
	Channel       *string       `db:"channel" json:"channel"`              //varchar 渠道
	Result        *int32        `db:"result" json:"result"`                //tinyint 结果(0 成功/1 失败)
	Deleted       *int32        `db:"deleted" json:"deleted"`              //tinyint 删除
	TenantId      *int64        `db:"tenant_id" json:"tenantId"`           //bigint 租户
	Creator       null.String   `db:"creator" json:"creator"`              //varchar 创建人
	CreateTime    null.DateTime `db:"create_time" json:"createTime"`       //datetime 创建时间
	Updater       null.String   `db:"updater" json:"updater"`              //varchar 更新人
	UpdateTime    null.DateTime `db:"update_time" json:"updateTime"`       //datetime 更新时间
}
